### PR TITLE
update pumgen package

### PR DESCRIPTION
**Pull request by Thomas-Ulrich**
- add scorec version 
- limit openmpi to 4
- allow larger hdf5 versions

### CODE REVIEW

File: spack/packages/pumgen/package.py [hunk 1]
```diff
@@ -8,30 +8,34 @@
 
 class Pumgen(CMakePackage):
     homepage = "https://github.com/SeisSol/PUMGen/wiki/How-to-compile-PUMGen"
+    git = "https://github.com/SeisSol/PUMGen.git"
     version('develop',
-            git='https://github.com/SeisSol/PUMGen.git',
             branch='master',
             submodules=True)
 
+    version('1.0.1', tag='v1.0.1', commit='13d1e1f41e3cdcb1d3c94f1870f003778e5f0ce0', submodules=True)
```
Comment (davschneller) [inline spack/packages/pumgen/package.py:16]: In principle, you'll need to enforce `+scorec` with version 1.0.1 .

**Review by davschneller — APPROVED** [review]
LGTM; just one short comment about v1.0.1

File: spack/packages/pumgen/package.py [hunk 2]
```diff
@@ -8,30 +8,42 @@
...
+    depends_on(f"openmpi@:4", when=f"+with_simmetrix ^[virtuals=mpi] openmpi")
+
     depends_on('netcdf-c +shared +mpi', when='+with_netcdf') # NOTE: only tested with 4.4.0 version
-    depends_on('hdf5@1.10:1.12.2 +shared +threadsafe +mpi')
-    depends_on('pumi +int64 +zoltan ~fortran', when='~with_simmetrix')
+    depends_on('hdf5@1.10: +shared +threadsafe +mpi')
     depends_on('simmetrix-simmodsuite', when='+with_simmetrix')
-    depends_on('pumi +int64 simmodsuite=base +zoltan ~fortran ~simmodsuite_version_check', when='+with_simmetrix')
+    depends_on('pumi +int64 simmodsuite=base +zoltan ~fortran ~simmodsuite_version_check', when='+with_simmetrix +scorec')
+    with when("@1.0.1"):
```
Comment (davschneller) [inline spack/packages/pumgen/package.py:30]: You could also go for a `requires("+scorec", when="@:1.0.1")` or similar, if you want to shorten the code a bit. Or you merge like it is now. :)

Comment (Thomas-Ulrich) [reply spack/packages/pumgen/package.py:30]: yes, but  this does not covers pumi without simmetrix for @1.0.1
(that means user would need to have a simmetrix license to install)

**Pull request merged**